clients/drcachesim/docs/drcachesim.dox.in

```diff
@@ -1364,13 +1369,31 @@ closely match the actual hardware if so desired.
 Traces also include markers indicating disruptions in user mode control
 flow such as signal handler entry and exit.
 
-Offline traces guarantee that a branch target instruction entry in a
-trace must immediately follow the branch instruction with no intervening
-thread switch.  This allows a core simulator to identify the target of a
-branch by looking at the subsequent trace entry.  This guarantee
-does not hold when a kernel event such as a signal is delivered
-immediately after a branch; however, each marker indicating such a kernel transfer
-includes the interrupted PC, explicitly providing the branch target.
+Offline traces explicitly identify whether each conditional branch was
+taken or not, and include a marker showing the target of unconditional
+branches, for convenience to avoid having to read either the
+subsequent entry or the kernel transfer event marker (or infer branch
+behavior for rseq aborts):
+
+```
+     2212815     1648444:     1249326 ifetch       6 byte(s) @ 0x00007f3406720707 48 3d 01 f0 ff ff    cmp    %rax, $0xfffff001
+     2212816     1648445:     1249326 ifetch       2 byte(s) @ 0x00007f340672070d 73 01                jnb    $0x00007f3406720710 (untaken)
+     2212817     1648445:     1249326 <marker: indirect branch target 0x7f34066a8b37>
+     2212818     1648446:     1249326 ifetch       1 byte(s) @ 0x00007f340672070f c3                   ret
+     2212819     1648446:     1249326 read         8 byte(s) @ 0x00007ffd91e24fa8 by PC 0x00007f340672070f
+     2212820     1648447:     1249326 ifetch       5 byte(s) @ 0x00007f34066a8b37 4c 8b 54 24 48       mov    0x48(%rsp), %r10
+```
+
+Offline traces further guarantee that a branch target instruction
```

Comment:
It is convenient to use that mechanism in raw2trace to compute the targets.  We could try to use buffers to accomplish the same thing but it will take more effort.  You're saying let's drop the documented guarantee, even if we leave the internal mechanism?  To give us flexibility in the future?

Reply:
Yes let's remove this guarantee that was added only when we needed it to know the branch's result.

I thought it might make the implementation easier too. But I haven't thought it through.

Reply:
Yes that does sound best.  I removed this whole paragraph.